src/main/java/com/lmax/disruptor/BatchEventProcessor.java

```diff
@@ -180,6 +198,11 @@ private void processEvents()
                     break;
                 }
             }
+            catch (final RewindableException e)
```

Comment:
In the event of a permanent exception, which may be classed as rewindable is there any idea how you're meant to track how many attempts it has at processing the batch?

Are you meant to encapsulate that in `RewindPauseStrategy::pause`?
I wonder if the `RewindableException e` should be passed to the pause strategy, and if it should be geared a little more towards more than just pausing, e.g. logging that exception?

Reply:
I have generified the RewindPauseStrategy to a RewindStrategy. The user is free to provide a custom strategy and decide whether the batch should rewind or delegate the exception to the exception handler. This will give it more usefulness than just pausing.